src/Admin/AbstractAdmin.php

```diff
@@ -3123,6 +3123,35 @@ private function buildRoutes(): void
             $extension->configureRoutes($this, $this->routes);
         }
     }
+
+    /**
+     * Determines default sorting of list (name of field and direction).
+     * Returns an array with 2 keys: "_sort_by" - name of field, "_sort_order" - sorting direction.
+     *
+     * Looks for sorting parameters in this order (first found will be used):
+     * 1. $this->$datagridValues property
+     * 2. Query used to load data
+     * 3. Defaults based on entity
+     */
+    private function determineSorting(): array
+    {
+        $defaults = $this->getModelManager()->getDefaultSortValues($this->getClass());
+        $query = $this->createQuery();
+        $datagrid = $this->datagridValues;
+
+        $querySortBy = null;
+        $querySortOrder = null;
+
+        if (null !== $query) {
```

Comment:
This check does not seems necessary since createQuery always return a ProxyQuery

Reply:
If `\Sonata\AdminBundle\Admin\AbstractAdmin::createQuery()` will be overridden, we don't know if it returns instance of `\Sonata\AdminBundle\Datagrid\ProxyQueryInterface`, because it hasn't return type declaration. This `if` is required to avoid `Fatal error Call to a member function on null`.

Maybe `public function createQuery($context = 'list');` should be changed to `public function createQuery($context = 'list'): ProxyQueryInterface;`. WDYT?

Reply:
The phpDoc say `ProxyQueryInterface`.
If someone override the method, he has to respect the phpDoc even if there is no strict return type. There is already multiple usages of $admin->createQuery without `null` checks.

Adding the return type would be a BC-break change. It'll be done in 4.0 ;)

Reply:
Sure, I understand it's not the best time for modification of return type declaration. Do you really need to remove the `if (null !== $query) {` condition? Stronger verification means less problems 🤓

Reply:
If we follow this, you should check that it's not a string, nor an int, ...

The ORM use the Query this way:
![image](https://user-images.githubusercontent.com/9052536/78505153-cc405780-7771-11ea-87ee-24a923a8b5ac.png)

And Datagrid does not accept null in his constructor
![image](https://user-images.githubusercontent.com/9052536/78505160-d9f5dd00-7771-11ea-8274-d82bea6e6cab.png)

So trust me, if someone return null in the createquery, he would have others problems ;)

I think it would be better to avoid useless check. If a static analysis like phpstan was running at a enough high level, it would return an error because a check `null === ProxyQueryInterface` is always false.

Reply:
No problem. It's done (424e9e1f commit).